src/sources/datadog/agent.rs

```diff
@@ -76,36 +115,117 @@ impl DatadogAgentSource {
     fn new(store_api_key: bool) -> Self {
         Self {
             store_api_key,
-            api_key_matcher: Regex::new(r"^/v1/input/(?P<api_key>[[:alnum:]]{32})/??")
-                .expect("static regex always compiles"),
+            api_key_matcher: Regex::new(
+                r"^(/v1/input|/api/v2/logs)/(?P<api_key>[[:alnum:]]{32})/??",
```

Comment:
in v2 api key can be passed as a header `dd-api-key` or as a query param(no need to support for agent), but not in the path.

Reply:
@vladimir-dd I pushed a change to parse the query param 😉

Reply:
Great, thanks.